plugins/katello/developers.md

```diff
@@ -229,37 +229,99 @@ Make sure to follow the instructions and update the docs to use the current rele
 
-First, we take the pinned Foreman modules in the Puppetfile from the most current release, e.g. https://github.com/theforeman/foreman-installer/blob/1.9-stable/Puppetfile.  Overwrite the Foreman section in katello-installer's Puppetfile with these locked down versions.
+Go through the [list of managed modules](https://github.com/theforeman/foreman-installer-modulesync/blob/master/managed_modules.yml).
 
-Next, we'll lock down the Katello modules. First, Check out all the puppet modules:
+```sh
+# Review the the changes since the last tag
+tig $(git describe --abbrev=0 --tags)..HEAD
 
-```bash
-for module in candlepin certs capsule common elasticsearch gutterball katello crane service_wait pulp
-do
-  git clone git@github.com:Katello/puppet-${module}.git
-done
-```
+# If it needs a major or minor version bump
+# Note: usually we already bumped it for patch releases but do check!
+bundle exec rake module:bump:major
+bundle exec rake module:bump:minor
 
-Review the modules for releasing - we should be keeping them up to date when making incompatible changes, but do any need bumps of the major or minor version? If so, do so now.
+# Update CHANGELOG.md
+# TODO: modulesync to include github_changelog_generator in Gemfile
+# See https://github.com/skywinder/Github-Changelog-Generator#github-token on using a token to avoid ratelimits
+github_changelog_generator -u theforeman -p $(basename $PWD) --unreleased-only -o new-changelog.md --future-release $(bundle exec rake module:version)
+
+# Now merge new-changelog.md into CHANGELOG.md - do note that it doesn't handle
+# branches so if anything was merged to a stable branch you should manually put
+# it in the correct release. This may also break badly on stable branches.
+
+# Create the release commit
+git add CHANGELOG.md
+VERSION=$(bundle exec rake module:version) && git commit -m "Release $VERSION"
 
-Consider the changes that have been made to the module since the last release, do we need a new release?
-Use [blacksmith](https://github.com/maestrodev/puppet-blacksmith) to push to forge.
+# Create a new tag
+# TODO: bundle exec rake module:tag doesn't create a GPG signed tag
+VERSION=$(bundle exec rake module:version) && git tag -s $VERSION -m "Version $VERSION"
 
-```bash
-rake module:release
+# Push the changes to github
+git push --follow-tags
+
+# Push the release to the forge
+# TODO: Automate this through Travis like voxpupuli?
+bundle exec rake module:push
+
+# Automated using pass - https://www.passwordstore.org/
+# Assumes you have a katello/forge or theforeman/forge password configured
+BLACKSMITH_FORGE_USERNAME="$(jq -r .author metadata.json | tr [A-Z] [a-z])" && BLACKSMITH_FORGE_USERNAME=$BLACKSMITH_FORGE_USERNAME BLACKSMITH_FORGE_PASSWORD=$(pass show $BLACKSMITH_FORGE_USERNAME/forge | head -n 1) bundle exec rake module:push
+
+# Start the next minor version
+# TODO: should we append -rc0 like voxpupuli?
```

Comment:
https://github.com/voxpupuli/voxpupuli-release-gem/blob/fd8f279de8dcae75f7bac7daed8e482aaedaed9a/lib/voxpupuli/release/rake_tasks.rb#L24-L27

Reply:
I don't know that we do enough robust lifecycle to warrant doing release candidates so I'd say no to doing this at present.

Reply:
I think it's more to have a placeholder ```version-next``` value. They're not actually released but it's about dealing with ```metadata.json``` having a ```version``` - it should be newer than what's released to show the difference but maybe it shouldn't be as new as the next release either (as the current code does).

Reply:
I see what you mean. I could be fine with that approach. @stbenjam given your experience with our mixture of bumping strategies between releases, any thoughts?

Reply:
In general, bumping just the minor has worked fine for us.  But, I like the rc0 approach.